src/modules/auth/auth.service.ts

```diff
@@ -257,4 +268,66 @@ export default class AuthenticationService {
       },
     };
   }
+
+  async requestSignInToken(requestSignInTokenDto: RequestSigninTokenDto) {
+    const { email } = requestSignInTokenDto;
+
+    const user = await this.userService.getUserRecord({ identifier: email, identifierType: 'email' });
+
+    if (!user) {
+      throw new NotFoundException({
+        message: USER_NOT_FOUND,
+        status_code: HttpStatus.NOT_FOUND,
+      });
+    }
+
+    const otpExist = await this.otpService.findOtp(user.id);
+
+    const newOtp = generateSixDigitToken();
+
+    if (otpExist) {
+      await this.otpService.deleteOtp(user.id);
+
+      await this.otpService.createOtp(user.id);
+
+      await this.emailService.sendLoginOtp(user.email, newOtp);
+
+      return;
+    }
+    await this.otpService.createOtp(user.id);
+
+    await this.emailService.sendLoginOtp(user.email, newOtp);
```

Comment:
```suggestion
if (otpExist) {
    await this.otpService.deleteOtp(user.id);
  }
  
  // Generate a new OTP and save it
  const newOtp = generateSixDigitToken();
  await this.otpService.createOtp(user.id);

  // Send the OTP to the user's email
  await this.emailService.sendLoginOtp(user.email, newOtp);
```

Reply:
Fixed

Reply:
@CFCIfe I have made the required changes. Help review.